[BUG] generated `_api.h` uses internal "CYTHON_

### Describe the bug

I've been exploring the limited API support (great so far!), and some packages (mpi4py, petsc4py) use a pattern like:

```cython
ctypedef api class TestClass [
    type   Test_Type,
    object TestObject,
]:
    cdef int attr
```

which emits a generated `_api.h` header. When this header is included (indirectly from their package header) to use the package's C API in downstream packages that _are not themselves Cython_, but the limited API is used, compilation fails with errors like:

```
│ │   In file included from $PREFIX/lib/python3.12/site-packages/mpi4py/include/mpi4py/mpi4py.h:26:
 │ │   $PREFIX/lib/python3.12/site-packages/mpi4py/include/mpi4py/../../MPI_api.h:191:40: error: member access into incomplete type 'PyTypeObject' (aka '_typeobject')
 │ │     191 |     itemsize = ((PyTypeObject *)result)->tp_itemsize;
 │ │         |                                        ^
 │ │   $PREFIX/include/python3.12/pytypedefs.h:20:16: note: forward declaration of '_typeobject'
 │ │      20 | typedef struct _typeobject PyTypeObject;
```

Looking into the code, it appears to be because the internal `CYTHON_COMPILING_IN_LIMITED_API` is not defined when `Py_LIMITED_API` is defined.



### Code to reproduce the behaviour:

```cython
ctypedef api class TestClass [
    type   Test_Type,
    object TestObject,
]:
    cdef int attr
```

which produces the relevant code:

```c
#if !CYTHON_COMPILING_IN_LIMITED_API
    basicsize = ((PyTypeObject *)result)->tp_basicsize;
    itemsize = ((PyTypeObject *)result)->tp_itemsize;
#else
```

but the limited API branch is _not_ taken when this header is included and `Py_LIMITED_API` is defined.

### Expected behaviour

limited api branch in generated `_api.h` is taken when Py_LIMITED_API is defined, without requiring what appears to be cython-internal `-DCYTHON_COMPILING_IN_LIMITED_API=1`.

### OS

macOS, linux

### Python version

3.12.9

### Cython version

3.1.3

### Additional context

I'm guessing that some subset of the macro definitions in the top of a regular Cython .c file should also be in the `_api.h`? Or maybe these checks should be directly for `Py_LIMITED_API`? Or maybe packages should be doing something different in exposing these headers to make sure the right things are defined when included from a non-Cython .c file? I haven't been able to find much documentation of how `cdef api` is supposed to be used to produce a C API for a Cython library like mpi4py/petsc4py.

--- a/Cython/Utility/ImportExport.c
+++ b/Cython/Utility/ImportExport.c
@@ -472,6 +472,8 @@ static PyTypeObject *__Pyx_ImportType_$cyversion(PyObject* module, const char *m
 /////////////// TypeImport ///////////////
 //@substitute: naming
 
+// Note that this goes into headers so CYTHON_COMPILING_IN_LIMITED_API isn't available.
+
 #ifndef __PYX_HAVE_RT_ImportType_$cyversion
 #define __PYX_HAVE_RT_ImportType_$cyversion
 static PyTypeObject *__Pyx_ImportType_$cyversion(PyObject *module, const char *module_name, const char *class_name,
@@ -480,7 +482,7 @@ static PyTypeObject *__Pyx_ImportType_$cyversion(PyObject *module, const char *m
     PyObject *result = 0;
     Py_ssize_t basicsize;
     Py_ssize_t itemsize;
-#if CYTHON_COMPILING_IN_LIMITED_API
+#ifdef Py_LIMITED_API
     PyObject *py_basicsize;
     PyObject *py_itemsize;
 #endif
@@ -494,7 +496,7 @@ static PyTypeObject *__Pyx_ImportType_$cyversion(PyObject *module, const char *m
             module_name, class_name);
         goto bad;
     }
-#if !CYTHON_COMPILING_IN_LIMITED_API
+#ifndef Py_LIMITED_API
     basicsize = ((PyTypeObject *)result)->tp_basicsize;
     itemsize = ((PyTypeObject *)result)->tp_itemsize;
 #else

--- a/tests/run/api_class.srctree
+++ b/tests/run/api_class.srctree
@@ -1,0 +1,55 @@
+PYTHON setup.py build_ext --inplace
+PYTHON -c "import useapi; useapi.make_testclass()"
+
+################# setup.py ######################
+
+# The tests in this file are mainly about ensuring that the
+# correct utility code is included when needed.
+# Therefore, future additions should add new files rather than
+# new code in the existing files.
+
+from setuptools import setup, Extension
+from Cython.Build import cythonize
+
+ext_modules = (
+    cythonize("mainmodule.pyx") +
+    cythonize(Extension("useapi", sources=["useapi.pyx", "use_api_c.c"]))
+)
+
+import sys
+print(ext_modules, file=sys.stderr)
+
+setup(ext_modules=ext_modules)
+
+################ mainmodule.pyx #################
+
+ctypedef api class TestClass [type Test_Type, object TestObject]:
+    cdef int attr
+
+############## use_api_c.c ################
+
+// This is making sure that mainmodule_api is functional
+// when used outside Cython (and with/without the Limited API
+// depending on how the tests are run).
+
+#include "mainmodule_api.h"
+
+PyObject *make_testclass_c(void) {
+    if (import_mainmodule() == -1) return NULL;
+    PyObject *tpl = PyTuple_New(0);
+    if (!tpl) return NULL;
+    PyObject *result = PyObject_Call((PyObject*)&Test_Type, tpl, NULL);
+    Py_DECREF(tpl);
+    return result;
+}
+
+############## useapi.pyx #######################
+
+cdef extern from *:
+    """
+    PyObject *make_testclass_c(void);
+    """
+    object make_testclass_c()
+
+def make_testclass():
+    return make_testclass_c()

